packages/nimble-components/src/table/components/row/index.ts

```diff
@@ -20,13 +21,21 @@ export class TableRow<
     @observable
     public data?: TData;
 
-    // TODO: Temporarily assume the set of columns will be an array of strings.
-    // Eventually, this will be an array of column definitions.
     @observable
-    public columns: string[] = [];
+    public columns: TableColumn[] = [];
 
-    public getCellValue(column: string): TableFieldValue {
-        return this.data ? this.data[column] : undefined;
+    public getCellValue(column: TableColumn): TableCellState<TData> {
+        const dataKeys = column.getRecordFieldNames();
+        const cellDataValues = dataKeys.map(key => this.data![key]);
+        const cellData = Object.fromEntries(
+            column.cellStateDataFieldNames.map((k, i) => [k, cellDataValues[i]])
+        );
+        const columnConfig = column.getColumnConfig?.() ?? {};
+        const cellState = {
+            data: cellData,
+            columnConfig
+        } as TableCellState<TData>;
```

Comment:
Didn't try to sync it down myself but can we avoid the `as` here? Seems like this type should be compatible with TableSellState right? Shouldn't need to lie to TS about it.

Reply:
Updated.